storage/src/nodestore/mod.rs

```diff
@@ -685,6 +613,83 @@ where
     }
 }
 
+impl<T, S: ReadableStorage> NodeStore<T, S> {
+    /// Read a [Node] from the provided [`LinearAddress`].
+    /// `addr` is the address of a `StoredArea` in the `ReadableStorage`.
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`FileIoError`] if the node cannot be read.
+    pub fn read_node_from_disk(
+        &self,
+        addr: LinearAddress,
+        mode: &'static str,
+    ) -> Result<SharedNode, FileIoError> {
+        if let Some(node) = self.storage.read_cached_node(addr, mode) {
+            return Ok(node);
+        }
+
+        debug_assert!(addr.is_aligned());
```

Comment:
really unrelated to your change, but since I saw it elsewhere:

this panics when branch factor 256 is enabled. obv low priority but I didn't linger enough to figure out why.